opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -467,6 +481,48 @@ static int mca_btl_tcp_component_close(void)
     return OPAL_SUCCESS;
 }
 
+/*
+ * Set an appropriate bandwidth for the btl instance. On Linux, we obtain this
+ * via an ioctl(). All other OSs, we return MCA_BTL_TCP_BTL_BANDWIDTH.
+ */
+static unsigned int
+mca_btl_tcp_ethtool_get_speed (const char *if_name)
+{
+#if !HAVE_DECL_SIOCETHTOOL
+    return MCA_BTL_TCP_BTL_BANDWIDTH;
+#else
+    int sockfd;
+    struct ifreq ifr;
+    struct ethtool_cmd edata = {
+        .cmd = ETHTOOL_GSET,
+    };
+    unsigned int speed = 0;
```

Comment:
Should speed be set to MCA_BTL_TCP_BTL_BANDWIDTH here, just in case you `goto out`?

I ask because in the non-SIOCETHTOOL case, you return MCA_BTL_TCP_BTL_BANDWIDTH.  But in the error cases, you seem to return 0 (i.e., where you `goto out` and if `ethtool_cmd_speed()` returns -1).

If you want to distinguish between returning an error and the returned speed, then we'll probably want to return the error code, and pass in a `uint32 *speed` parameter to return the speed.